File: Shared/Blog.Contracts/V1/Validators/UsersValidators/UpdateProfileRequestValidator.cs
using Blog.Contracts.V1.Requests.UsersRequests;
using FluentValidation;

namespace Blog.Contracts.V1.Validators.UsersValidators;

/// <summary>
/// Change password request validator.
/// </summary>
/// <seealso cref="AbstractValidator{RegistrationRequest}" />
public class UpdateProfileRequestValidator : AbstractValidator<UpdateProfileRequest>
{
    public UpdateProfileRequestValidator()
    {
        RuleFor(x => x.Email)
            .NotEmpty()
            .EmailAddress();

        RuleFor(x => x.FirstName)
            .NotEmpty()
            .Matches("^[a-zA-Z]*S")
            .MinimumLength(2)
            .MaximumLength(64);

        RuleFor(x => x.LastName)
            .NotEmpty()
            .Matches("^[a-zA-Z]*S")
            .MinimumLength(2)
            .MaximumLength(64);

        RuleFor(x => x.PhoneNumber)
            .Matches(@"(?:(?:(\s*\(?([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\s*)|([2-9]1[02-9]|[2‌​-9][02-8]1|[2-9][02-8][02-9]))\)?\s*(?:[.-]\s*)?)([2-9]1[02-9]|[2-9][02-9]1|[2-9]‌​[02-9]{2})\s*(?:[.-]\s*)?([0-9]{4})");
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/UsersValidators/UpdateProfileRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.UsersValidators

open FluentValidation
open System
open System.Linq.Expressions
open Blog.FSharp.Contracts.V1.Requests.UsersRequests

/// <summary>
/// Update profile request validator.
/// </summary>
type UpdateProfileRequestValidator() =
    inherit AbstractValidator<UpdateProfileRequest>() 

    do
        // Email must not be empty and must be in a valid email format
        base.RuleFor(fun x -> x.Email)
            .NotEmpty()
            .EmailAddress()
        |> ignore

        // FirstName must not be empty, must contain only letters,
        // and must be between 2 and 64 characters long
        base.RuleFor(fun x -> x.FirstName)
            .NotEmpty()
            .Matches("^[a-zA-Z]*$")
            .MinimumLength(2)
            .MaximumLength(64)
        |> ignore

        // LastName must not be empty, must contain only letters,
        // and must be between 2 and 64 characters long
        base.RuleFor(fun x -> x.LastName)
            .NotEmpty()
            .Matches("^[a-zA-Z]*$")
            .MinimumLength(2)
            .MaximumLength(64)
        |> ignore

        // PhoneNumber must match a valid phone number pattern (+ optional, 10-15 digits)
        base.RuleFor(fun x -> x.PhoneNumber)
            .Matches(@"^\+?\d{10,15}$")
        |> ignore